CribblyBackend/Services/GameService.cs

```diff
@@ -0,0 +1,82 @@
+using System.Data;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CribblyBackend.Models;
+using Dapper;
+
+namespace CribblyBackend.Services
+{
+    public interface IGameService
+    {
+        Task<Game> GetById(int Id);
+        void Update(Game Game);
+        Task Create(Game Game);
+        void Delete(Game Game);
+    }
+    public class GameService : IGameService
+    {
+        IDbConnection connection;
+        public GameService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public async Task<Game> GetById(int id)
+        {
+            var players = new Dictionary<int, Player>();
+            var teams = new Dictionary<int, Team>();
+            var game = (await connection.QueryAsync<Game, Team, Player, Game>(
+                @"
+                    SELECT * FROM Assignments a 
+                    LEFT JOIN Games g on a.GameId = g.Id 
+                    LEFT JOIN Teams t on a.TeamId = t.Id 
+                    LEFT JOIN Players p on p.TeamId = t.Id
+                    WHERE GameId = @id
+                ", 
+                (g, t, p) =>
+                {
+                    if (!players.TryGetValue(p.Id, out Player _))
+                    {
+                        p.Team = new Team(){Id = t.Id};
```

Comment:
🤔 I don't think this line is necessary since `p` will always get overwritten and it's not used elsewhere in this scope

Reply:
I use this on line 56 to separate all of the players into their respective teams. I believe without it, the app will not know which players are on which team:

`team.Players = players.Values.Where(p => **p.Team.Id** == team.Id).ToList();`

If we can do this without it though, I am all for simplicity!

Reply:
They're named the same thing, but the one on L56 is a different variable (local to the anonymous function being passed into the `Where` call). Variables are only known within their scope (a set of curly braces) which is why you're allowed to defined two different variables with the same name.